### PR TITLE
test: add integration tests for the cli's structure

### LINKED ISSUE #2: Create some introductory E2E tests
## Summary
We should go ahead and add some simple E2E tests that establish how we create them moving forward. As a first pass, it likely makes sense to make a few tests that guarantee the flags being passed are working as intended.

### PR DESCRIPTION
**Pull request by tylerslaton**
## Summary
Adding a few introductory integration tests to ensure we are testing our code moving forward.

Closes #2

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,6 @@ docker-api = "0.9.1"
 tokio = { version = "1.0", features = ["full"] }
 futures-util = "0.3"
 tar = "0.4"
+predicates = "2.1.1"
+assert_cmd = "2.0.4"
+rand = "0.8.5"
```
Comment (exdx) [inline Cargo.toml:16]: We might be able to add these under `dev-dependencies` since they are only for testing.

File: tests/cli.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
+use assert_cmd::Command;
```
Comment (exdx) [inline tests/cli.rs:1]: Very impressive! Nice set of e2e tests that exercise the current functionality.

**Review by exdx — APPROVED** [review]
LGTM

I had one comment about the dependencies, but other that this is good to go.

CI is red however, once that's resolved let's merge.

**Comment by exdx** [comment]
Seems like we need to add quay.io to the image under test? `tyslaton/sample-catalog:v0.0.4` -- or is this on dockerhub?

**Comment by tylerslaton** [comment]
> Seems like we need to add quay.io to the image under test? `tyslaton/sample-catalog:v0.0.4` -- or is this on dockerhub?

We talked about this a bit offline but there seems to be a bug with the root binary not being able to create a container out of the image it pulls. To answer the original question, it is on both repository sites so the image definitely exists and is pullable.

File: Cargo.toml (hunk 1 above)
Comment (tylerslaton) [reply Cargo.toml:18]: Yep! Went ahead and fix this.

File: src/lib.rs [hunk 3]
```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
     })
 }
 
+// Returns a Result with an Image struct and an Error (if one occured)
+fn split_repo_and_tag(full_image: String) -> Image {
+    let image_split: Vec<&str> = full_image.split(":").collect();
+
+    let mut repo = String::from("");
```
Comment (exdx) [inline src/lib.rs:94]:
```suggestion
    let mut repo = String::new()
```

File: src/lib.rs [hunk 4]
```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
...
+    if image_split.get(0) != None {
```
Comment (exdx) [inline src/lib.rs:95]:
```suggestion
    if let Some(image_split.get(0) {
```

File: src/lib.rs [hunk 5]
```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
...
+        repo = image_split[0].to_string();
+    }
+
+    let mut tag = String::from("");
```
Comment (exdx) [inline src/lib.rs:99]: see above

File: src/lib.rs [hunk 6]
```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
...
+    if image_split.get(1) != None {
```
Comment (exdx) [inline src/lib.rs:100]: see above

File: src/lib.rs [hunk 7]
```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
     })
 }
 
+// Returns a Result with an Image struct and an Error (if one occured)
+fn split_repo_and_tag(full_image: String) -> Image {
```
Comment (exdx) [inline src/lib.rs:91]: could we make function part of the Image struct? Using `Impl Image` we define the types methods, and we can add this one there.

**Pull request merged**